To: vim_dev@googlegroups.com
Subject: Patch 7.3.217
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.217
Problem:    Inside an "if" a ":wincmd" causes problems.
Solution:   When skipping commands let ":wincmd" skip over its argument.
Files:      src/ex_docmd.c


*** ../mercurial/vim73/src/ex_docmd.c	2011-05-25 12:51:17.000000000 +0200
--- src/ex_docmd.c	2011-06-13 01:15:17.000000000 +0200
***************
*** 2595,2600 ****
--- 2595,2601 ----
  	    case CMD_unlet:
  	    case CMD_verbose:
  	    case CMD_vertical:
+ 	    case CMD_wincmd:
  				break;
  
  	    default:		goto doend;
***************
*** 8237,8243 ****
      p = skipwhite(p);
      if (*p != NUL && *p != '"' && eap->nextcmd == NULL)
  	EMSG(_(e_invarg));
!     else
      {
  	/* Pass flags on for ":vertical wincmd ]". */
  	postponed_split_flags = cmdmod.split;
--- 8238,8244 ----
      p = skipwhite(p);
      if (*p != NUL && *p != '"' && eap->nextcmd == NULL)
  	EMSG(_(e_invarg));
!     else if (!eap->skip)
      {
  	/* Pass flags on for ":vertical wincmd ]". */
  	postponed_split_flags = cmdmod.split;
*** ../vim-7.3.216/src/version.c	2011-06-13 01:07:22.000000000 +0200
--- src/version.c	2011-06-13 01:18:48.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     217,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
173. You keep tracking down the email addresses of all your friends
     (even childhood friends).

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
